src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -39,6 +39,8 @@ public partial class Activity : IDisposable
         private static readonly IEnumerable<ActivityLink> s_emptyLinks = new ActivityLink[0];
         private static readonly IEnumerable<ActivityEvent> s_emptyEvents = new ActivityEvent[0];
 #pragma warning restore CA1825
+        internal static readonly Random s_random = new Random();
```

Comment:
I don't think random is thread safe https://github.com/dotnet/runtime/issues/43887.

Reply:
It's not. (Often this manifests as zeros starting to be returned from all accesses after some period of time / some number of collisions.)

Reply:
Is there a safe way to generate random numbers then? or should I look at different solution here?

Reply:
One typical solution employed is to have a ThreadStatic Random instance.  Or if you don't expect a lot of contention, you can also lock around access to the instance.

Reply:
>  (Often this manifests as zeros starting to be returned from all accesses after some period of time / some number of collisions.)

Is this generic issue in Random class? or this happen only with calling it from multiple threads in same time?

Reply:
I would guess the cost of a thread-static access is small relative to the cost of the random number generation. Haven't benchmarked myself but I am guess single-digit ns for thread-static read, double-digit ns for creating random data.

> Is this generic issue in Random class? or this happen only with calling it from multiple threads in same time?

I certainly hope it is only an issue when being unsafely shared for multi-threaded use, otherwise it sounds like Random would be quite broken.

Reply:
> Is this generic issue in Random class? or this happen only with calling it from multiple threads in same time? 

When erroneously accessed by multiple threads concurrently.